### ISSUE TITLE
export-esp.sh does not set CLANG_PATH which can confuse bindgen (and any crate using clang-sys)

### ISSUE THREAD

**Issue by jasta** [esp-rs/espup#164]
## Bug description

export-esp.sh is missing the CLANG_PATH variable which is used by clang-sys to locate the clang runtime.  bindgen uses this to determine -isystem.  Without it, the host's include path will be used while the target's clang toolchain is used.

- Would you like to work on a fix? Yes

## To Reproduce

```
. ~/export-esp.sh
export TARGET_CC=riscv32-esp-elf-gcc
export TARGET_CXX=riscv32-esp-elf-g++
export TARGET_AR=riscv32-esp-elf-ar
cd <some-esp32-project>
cargo add lvgl --git https://github.com/jasta/lvgl-rs --branch esp32-fixup
cargo add lvgl-sys --git https://github.com/jasta/lvgl-rs --branch esp32-fixup
cargo add lvlg-codegen --git https://github.com/jasta/lvgl-rs --branch esp32-fixup
# Download https://github.com/rafaelcaricio/lvgl-rs/blob/master/lvgl-sys/vendor/include/lv_conf.h and store in CWD.
DEP_LV_CONFIG_PATH=$(pwd) cargo +nightly build --target riscv32imc-esp-espidf
```

Lots of build errors related to stdint.h types missing (like int16_t).  Similar problems are likely to occur with other crates utilizing bindgen, but not always consistently.  It may also repro differently depending on the host and target clang mismatches for the user testing it.

## Expected behavior

Should compile :)

## Screenshots

N/A

## Environment

- OS: Ubuntu 22.04
- espup version: 0.2.5

## Additional context

Part of a larger body of work to fix cc and bindgen (a couple other issues block general purpose use cases)

**Comment by jasta** [comment]
Not sure how exactly to make this work on Windows given: https://github.com/esp-rs/espup/blob/b9aa3e266c132f68f433db401d32bf2a0a358cd8/src/toolchain/llvm.rs#L50 ?  Kinda strange to me that they would be different like this?

**Comment by SergioGasquez** [comment]
> Not sure how exactly to make this work on Windows given: https://github.com/esp-rs/espup/blob/b9aa3e266c132f68f433db401d32bf2a0a358cd8/src/toolchain/llvm.rs#L48-L54 ?  Kinda strange to me that they would be different like this?

It's a strange requirement for windows: https://rust-lang.github.io/rust-bindgen/requirements.html#windows

**Comment by georgik** [comment]
@SergioGasquez  In case of Windows, it's necessary to set path to DLL which has slightly different location than on *nixes https://github.com/esp-rs/rust-build/blob/main/Install-RustToolchain.ps1#L48

**Comment by jasta** [comment]
I think I figured this out, the lib path being the bin path for Windows was a red herring.  That said, I ran into another snag which is that if the minified option is used for espup then only the clang libraries (not binaries) will be installed. 
 This will break cross compilation because bindgen indirectly invokes the clang binary here:

https://github.com/rust-lang/rust-bindgen/blob/a7ff8e136027c9542f65eca97a7ff3c0b870518d/bindgen/lib.rs#L2549

That goes on to apply a set of heuristics looking in CLANG_PATH, then generally anywhere in PATH, for a binary it can run.  Once it finds one, it does things like `clang -E -x c - -v -target whatever ...` to figure out what the include paths and other things should be.

I think I'll just conditionally the output CLANG_PATH.  If you ask for the minified installation, things just won't work out of the box quite like you expect :\
